[INJIWEB-1774] - Update XSRF token to be true by default

kongadurgesh opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Configurable CSRF protection enabled by default with a global ignore list and automatic CSRF token issuance via cookie for safe (GET) requests.

* **API**
  * OpenAPI updated to document X-XSRF-TOKEN header for multiple state-changing endpoints and guidance on obtaining the token.

* **Tests**
  * Added comprehensive tests covering CSRF token lifecycle, cookie behavior, HTTP methods, and config parsing.

* **Documentation**
  * Postman collection extended to extract/send CSRF tokens.

* **Chores**
  * Removed legacy third-party generation script; updated checksums and ignore entries.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kongadurgesh committed:
[INJIWEB-1774] - Add CSRF Token Implementation for Mimoto

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
[INJIWEB-1774] - Remove misleading comment

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
[INJIWEB-1774] - Update Value Param

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
[INJIWEB-1774] - Update tests method names
remove setAccesible from constructor

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
[INJIWEB-1776] - Delete legacy code (#990)

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

jackjain commented on src/main/java/io/mosip/mimoto/config/Config.java:
```diff
@@ -38,13 +50,22 @@
     private String baseUrl;
 
-    @Value("${mosip.security.csrf-enable:false}")
+    @Value("${mosip.security.csrf-enable}")
```
lets remove this configuration, CSRF will be enabled by default and driven via the csrf-ignore-urls

kongadurgesh replied:
added default configuration

jackjain commented:
- Will need a updated postman collection so that header gets automatically mapped from the cookie value
- Will need a demo with QAs since the change might impact them

kongadurgesh committed:
[INJIWEB-1774] - Add mimoto endpoints to ignoreUrls
Update default configuration

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

jackjain commented on src/main/java/io/mosip/mimoto/config/Config.java:
```diff
@@ -71,9 +95,15 @@ public Map<String, String> injiConfig() {
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http, SessionRepository sessionRepository) throws Exception {
-        if (!isCSRFEnable) {
+        if (isCSRFEnable) {
```
Lets remove this config, it will enabled by default and it will be driven completely by ignore csrf urls config.

kongadurgesh replied:
Removed the conditional check for CSRF

jackjain commented on src/main/resources/application-default.properties:
```diff
@@ -179,6 +179,16 @@ mosip.security.ignore-auth-urls=/safetynet/**,/actuator/**,/swagger-ui/**,/v3/ap
+mosip.security.csrf-enable=true
+# List of URLs that are excluded from CSRF protection. These URLs can still require authentication.
+# OAuth2 endpoints and public endpoints are typically excluded from CSRF protection.
+mosip.security.csrf-ignore-urls=/oauth2/**,/logout,/safetynet/**,/actuator/**,/swagger-ui/**,/v3/api-docs/**,/allProperties,/credentials/**,/credentialshare/**,/binding-otp,/wallet-binding,/get-token/**,/issuers,/issuers/**,/authorize,/req/otp,/vid,/req/auth/**,/req/individualId/otp,/aid/get-individual-id,/verifiers,/auth/*/token-login,/users/me,/wallets,/wallets/**
```
Lets check if docker works fine, lets add config there too

kongadurgesh replied:
added the ignore urls in docker setup

kongadurgesh committed:
[INJIWEB-1774] - Enable CSRF by defualt

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
[INJIWEB-1774] - Update Postman Collection
Add postcript for get wallets and get issuers endpoint
Add property to mimoto default properties  to docker setup

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
[INJIWEB-1774] - Update API Documentation

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
[INJIWEB-1774] - Remove dead config
Remove /wallets from ignore urls

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
[INJIWEB-1774] - remove client id and secret

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
Revert "[INJIWEB-1774] - Remove dead config"

This reverts commit 4ef6b44fb6e4330c880ac638c45d38fb17f87086.

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh committed:
[INJIWEB-1774] - Remove dead config
Remove /wallets from ignore urls

Signed-off-by: kongadurgesh <kongadurgesh20@gmail.com>

kongadurgesh commented on src/main/java/io/mosip/mimoto/config/Config.java:
```diff
@@ -53,6 +70,9 @@ public class Config {
     @Value("${mosip.security.ignore-auth-urls}")
     private String[] ignoreAuthUrls;
 
+    @Value("${mosip.security.csrf-ignore-urls}")
+    private String[] csrfIgnoreUrls;
```
The variable csrfIgnoreUrls comes from properties file, we dont expect it to be passed as empty strings
